tests/admin_changelist/tests.py

```diff
@@ -48,6 +48,7 @@ class ChangeListTests(TestCase):
 
     def setUp(self):
         self.factory = RequestFactory()
+        self.superuser = User.objects.create_superuser(username='super', email='a@b.com', password='xxx')
```

Comment:
Seems sensible based on the [example](https://docs.djangoproject.com/en/1.11/topics/testing/advanced/#example) in the documentation.

The user doesn't need to be a superuser, and I would suggest it is probably better they are not.
Just create a normal staff user:

```python
self.user = User.objects.create_user(username='staff', is_staff=True)
```

Reply:
It was just to avoid needing to add the permissions to the user (since it's used so many times)

Reply:
This also seems strange when I can see a `_create_superuser()` method immediately below `setUp()`.

Are we sure we are doing the correct thing here? Given these tests work in the existing code where there are add, change and delete permissions; what makes adding a view permission require all these changes. We should pinpoint what bit of the new code is causing this and check it.

Further, if we do need to do this, we should probably also be using `_mocked_authenticated_request()`, also below.

Reply:
> Are we sure we are doing the correct thing here? Given these tests work in the existing code where there are add, change and delete permissions; what makes adding a view permission require all these changes. We should pinpoint what bit of the new code is causing this and check it.

Adding the `request.user` instance to each test is correct. 

Prior to this change, `changelist_view` checks the permissions before calling, `get_changelist_instance()`, which never uses `request.user`, since it's the same for everyone.

https://github.com/django/django/blob/acf9d6404545c58522c3011348d071d6bfca8f69/django/contrib/admin/options.py#L1594-L1598

Now, `get_actions()`, which is called by `get_changelist_instance()`, itself needs to check permissions, so `request.user` becomes requisite. 

A little refactoring (as a separate commit) — to re-use or move the helper methods is probably merited.